core/src/control/load_balancing/load_balancing_base.tpp

```diff
@@ -62,13 +78,16 @@ void LoadBalancingBase<TimeStepping>::initialize() {
   LOG(TRACE) << "LoadBalancingBase::initialize()";
 
   TimeSteppingScheme::TimeSteppingScheme::initialize();
+  timeSteppingScheme_.setUniqueDataPrefix(
+      StringUtility::optionalConcat(this->uniqueDataPrefix_, "load_balancing"));
   timeSteppingScheme_.initialize();
 }
 
 template <typename TimeStepping> void LoadBalancingBase<TimeStepping>::run() {
   initialize();
 
-  advanceTimeSpan();
+  auto checkpointing = this->context_.getCheckpointing();
```

Comment:
This code is not safe. Assuming a shared_ptr is returned then there must be a assertion check if the pointer is valid or not. Passing nullptr without checks will make it painfully difficult to track potential bugs.